client/checkout/classic/upe-deferred-intent-creation/stripe-checkout.js

```diff
@@ -0,0 +1,242 @@
+ * retrieves the necessary data from the UPE configuration and initializes the appearance. It then creates the
+ * Stripe elements and the Stripe payment element, which is attached to the gatewayUPEComponents object afterward.
+ *
+ * @param {Object} api The API object used to create the Stripe payment element.
+ * @param {string} paymentMethodType The type of Stripe payment method to create.
+ * @return {Object} A promise that resolves with the created Stripe payment element.
+ */
+async function createStripePaymentElement( api, paymentMethodType ) {
+	const amount = Number( getUPEConfig( 'cartTotal' ) );
+	const options = {
+		mode: 1 > amount ? 'setup' : 'payment',
+		currency: getUPEConfig( 'currency' ).toLowerCase(),
+		amount: amount,
+		paymentMethodCreation: 'manual',
+		paymentMethodTypes: [ paymentMethodType ],
+		appearance: initializeAppearance( api ),
+	};
+
+	const elements = api.getStripe().elements( options );
+	const createdStripePaymentElement = elements.create( 'payment', {
+		...getUpeSettings(),
+		wallets: {
+			applePay: 'never',
+			googlePay: 'never',
+		},
+	} );
+
+	gatewayUPEComponents[ paymentMethodType ].elements = elements;
+	gatewayUPEComponents[
+		paymentMethodType
+	].upeElement = createdStripePaymentElement;
+	return createdStripePaymentElement;
+}
+
+/**
+ * Mounts the existing Stripe Payment Element to the DOM element.
+ * Creates the Stipe Payment Element instance if it doesn't exist and mounts it to the DOM element.
+ *
+ * @param {Object} api The API object.
+ * @param {string} domElement The selector of the DOM element of particular payment method to mount the UPE element to.
+ **/
+export async function mountStripePaymentElement( api, domElement ) {
+	try {
+		if ( ! fingerprint ) {
+			const { visitorId } = await getFingerprint();
+			fingerprint = visitorId;
+		}
+	} catch ( error ) {
+		showErrorCheckout( error.message );
+		return;
+	}
+	const paymentMethodType = domElement.dataset.paymentMethodType;
+	const upeElement =
+		gatewayUPEComponents[ paymentMethodType ].upeElement ||
+		( await createStripePaymentElement( api, paymentMethodType ) );
+	upeElement.mount( domElement );
+}
+
+/**
+ * Handles the checkout process for the provided jQuery form and Stripe payment method type. The function blocks the
+ * form UI to prevent duplicate submission and validates the Stripe elements. It then creates a Stripe payment method
+ * object and appends the necessary data to the form for checkout completion. Finally, it submits the form and prevents
+ * the default form submission from WC Core.
+ *
+ * @param {Object} api The API object used to create the Stripe payment method.
+ * @param {Object} jQueryForm The jQuery object for the form being submitted.
+ * @param {string} paymentMethodType The type of Stripe payment method being used.
+ * @return {boolean} return false to prevent the default form submission from WC Core.
+ */
+let hasCheckoutCompleted;
```

Comment:
```suggestion
let hasCheckoutCompleted = false;
```
Not a huge deal, but if we don't need to check for this variable's `undefined` state, why not be more explicit and just declare this as `false` from the beginning?

Reply:
As discussed f2f, we'll leave it as it is due to some side-effects this small change brings.